tests/src/App/IndexedName.cpp

```diff
@@ -0,0 +1,421 @@
+
+    // Assert
+    EXPECT_STREQ(indexedName.getType(), "");
+}
+
+TEST_F(IndexedNameTest, nameAndTypeListConstructionWithoutAllowOthers)
+{
+    // Act
+    auto indexedName = Data::IndexedName("EDGE19", allowedTypes, false);
+
+    // Assert
+    EXPECT_STREQ(indexedName.getType(), "EDGE");
+    EXPECT_EQ(indexedName.getIndex(), 19);
+
+    // Act
+    indexedName = Data::IndexedName("EDGES_ARE_REALLY_GREAT19", allowedTypes, false);
+
+    // Assert
+    EXPECT_STREQ(indexedName.getType(), "");
+    EXPECT_EQ(indexedName.getIndex(), 19);
+
+    // Act
+    indexedName = Data::IndexedName("NOT_IN_THE_LIST42", allowedTypes, false);
+
+    // Assert
+    EXPECT_STREQ(indexedName.getType(), "");
+}
+
+TEST_F(IndexedNameTest, nameAndTypeListConstructionWithAllowOthers)
+{
+    // Act
+    auto indexedName = Data::IndexedName("NOT_IN_THE_LIST42", allowedTypes, true);
+
+    // Assert
+    EXPECT_STREQ(indexedName.getType(), "NOT_IN_THE_LIST");
+    EXPECT_EQ(indexedName.getIndex(), 42);
+}
+
+TEST_F(IndexedNameTest, byteArrayConstruction)
+{
+    // Arrange
+    QByteArray qba{"EDGE42"};
+
+    // Act
+    auto indexedName = Data::IndexedName(qba);
+
+    // Assert
+    EXPECT_STREQ(indexedName.getType(), "EDGE");
+    EXPECT_EQ(indexedName.getIndex(), 42);
+}
+
+TEST_F(IndexedNameTest, copyConstruction)
+{
+    // Arrange
+    auto indexedName = Data::IndexedName("EDGE42");
+
+    // Act
+    auto indexedNameCopy {indexedName};
+
+    // Assert
+    EXPECT_EQ(indexedName, indexedNameCopy);
+}
+
+TEST_F(IndexedNameTest, streamInsertionOperator)
+{
+    // Arrange
+    auto indexedName = Data::IndexedName("EDGE42");
+    std::stringstream ss;
+
+    // Act
+    ss << indexedName;
+
+    // Assert
+    EXPECT_EQ(ss.str(), std::string{"EDGE42"});
+}
+
+TEST_F(IndexedNameTest, compoundAssignmentOperator)
+{
+    // Arrange
+    constexpr int base{42};
+    constexpr int offset{10};
+    auto indexedName = Data::IndexedName("EDGE",base);
+
+    // Act
+    // Only += is defined for this class
+    indexedName += offset;
+
+    // Assert
+    EXPECT_EQ(indexedName.getIndex(), 52);
+}
+
+TEST_F(IndexedNameTest, preincrementOperator)
+{
+    // Arrange
+    auto indexedName = Data::IndexedName("EDGE42");
+
+    // Act
+    ++indexedName;
+
+    // Assert
+    EXPECT_EQ(indexedName.getIndex(), 43);
+}
+
+TEST_F(IndexedNameTest, predecrementOperator)
+{
+    // Arrange
+    auto indexedName = Data::IndexedName("EDGE42");
+
+    // Act
+    --indexedName;
+
+    // Assert
+    EXPECT_EQ(indexedName.getIndex(), 41);
+}
+
+TEST_F(IndexedNameTest, comparisonOperators)
+{
+    // Arrange
+    auto indexedName1 = Data::IndexedName("EDGE42");
+    auto indexedName2 = Data::IndexedName("EDGE42");
+
+    // Assert
+    EXPECT_EQ(indexedName1.compare(indexedName2), 0);
+    EXPECT_TRUE(indexedName1 == indexedName2);
+    EXPECT_FALSE(indexedName1 != indexedName2);
+    EXPECT_FALSE(indexedName1 < indexedName2);
```

Comment:
A few possiblities for the block starting on 214. Either mark the block comment as `// Act & Assert`. Which is reasonable. Or parlay it out like

```cpp
    // Act
    auto compareResult = indexedName1.compare(indexedName2)
    auto eqResult = (indexedName1 == indexedName2);
    auto neResult = (indexedName1 != indexedName2);
    auto ltResult = (indexedName1 < indexedName2);

    // Assert
    EXPECT_EQ(compareResult, 0);
    EXPECT_TRUE(eqResult);
    EXPECT_FALSE(neResult);
    EXPECT_FALSE(ltResult);
```

Either is good IMO.